### PR TITLE
AccessData Integration Update

### PR DESCRIPTION
**Pull request by ThomasVieth**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description
Rewrite the current integration to provide interaction the new FTK API. 

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 5.5.0
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [x] Yes
       - Further details:
       REST API endpoints and parameters have changed, alongside a new Python SDK is available which has been contained within a dockerfile to make this integration easier.
   - [ ] No

### CODE REVIEW

File: Packs/Accessdata/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+##### What does this pack do?
+
+Exterro’s FTK Connect features a robust API that enables a secure connection between Cortex XSOAR and FTK Enterprise.
+When Cortex XSOAR detects an attack, it sends an alert that is received by the FTK Connect API, which initiates the playbook where FTK Enterprise automatically initiates a collection job on the endpoint.
+
+Automatically preserve electronic evidence by initiating the immediate collection of data at the designated endpoint, using automated incident playbooks.
+Ensure evidence is collected and preserved in a legally defensible manner for full data integrity with forensically sound collection capabilities using a single, secure back-end database.
+Reduce data movement between platforms to avoid data spoliation with seamless handoff from incident detection to analysis and response. 
+
+Add helpful, relevant links below
```
Comment (mchasepan) [inline Packs/Accessdata/README.md:14]: Delete this line

File: Packs/Accessdata/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+Exterro’s FTK Connect features a robust API that enables a secure connection between Cortex XSOAR and FTK Enterprise.
+When Cortex XSOAR detects an attack, it sends an alert that is received by the FTK Connect API, which initiates the playbook where FTK Enterprise automatically initiates a collection job on the endpoint.
+
+Automatically preserve electronic evidence by initiating the immediate collection of data at the designated endpoint, using automated incident playbooks.
+Ensure evidence is collected and preserved in a legally defensible manner for full data integrity with forensically sound collection capabilities using a single, secure back-end database.
+Reduce data movement between platforms to avoid data spoliation with seamless handoff from incident detection to analysis and response. 
+
+Add helpful, relevant links below 
+
+- Learn more about FTK Enterprise: https://www.exterro.com/ftk-enterprise
```
Comment (mchasepan) [inline Packs/Accessdata/README.md:16]: Make these markdown links?

File: Packs/Accessdata/README.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
+New and advanced forms of cybersecurity threats are continually evolving to target enterprises.  In the event of a cybersecurity incident, rapid evidence capture and response is the key to quickly patching an enterprise’s vulnerabilities and preventing another attack.  With the integration between Cortex XSOAR, Exterro FTK Connect and FTK Enterprise, users can now leverage Cortex XSOARs security orchestration and automation capabilities to trigger the immediate capture and preservation of endpoint evidence by FTK Enterprise, which is crucial for incident investigation and recovery.
```
Comment (mchasepan) [inline Packs/Accessdata/README.md:1]:
```suggestion
New and advanced forms of cybersecurity threats are continually evolving to target enterprises.  In the event of a cybersecurity incident, rapid evidence capture and response is the key to quickly patching an enterprise’s vulnerabilities and preventing another attack.  With the integration between Cortex XSOAR, Exterro FTK Connect and FTK Enterprise, users can now leverage Cortex XSOAR's security orchestration and automation capabilities to trigger the immediate capture and preservation of endpoint evidence by FTK Enterprise, which is crucial for incident investigation and recovery.

```

File: Packs/Accessdata/README.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+- Learn more about the FTK Connect API: https://www.exterro.com/ftk-api
+
+2 Major Use Cases:
+
+1.   Data Exfiltration
+Once potential data exfiltration is identified from inside the network, a Cortex XSOAR playbook can automatically trigger the Exterro restful API to instantly capture a complete list of all processes currently running on the suspected machine, and their associated DLL files.  
+From here, you can identify any processes that are unauthorized, and invoke Exterro to automatically remediate them.
+An automatic job collection is then initiated by FTK Enterprise at this specified endpoint.
+The Cortex XSOAR and Exterro integration automates this time-intensive process that would normally need to be completed manually
```
Comment (mchasepan) [inline Packs/Accessdata/README.md:26]:
```suggestion
The Cortex XSOAR and Exterro integration automates this time-intensive process that would normally need to be completed manually.

```

File: Packs/Accessdata/README.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+2.   Automated Memory Dump
+Acquiring a memory dump or drive scan from a compromised endpoint in a suspected internal breach is key, especially before the evidence is lost
```
Comment (mchasepan) [inline Packs/Accessdata/README.md:29]:
```suggestion
Acquiring a memory dump or drive scan from a compromised endpoint in a suspected internal breach is key, especially before the evidence is lost.

```

**Comment by guykeller** [comment]
Hi there!
Due to a sequence of holidays in recent weeks, your review was delayed. We apologize for the inconvenience and will try to do our best to complete your review as quickly and efficiently as possible, please bear with us.
Thanks for your understanding, and for your contribution.

**Comment by ShahafBenYakir** [comment]
Hi @ThomasVieth Please start by filling the [contribution form](https://github.com/demisto/content/pull/15060#issuecomment-927712073). Them please address the rest of the comments on this PR:

**Comment by daryakoval** [comment]
@ThomasVieth Hi, Are you done? can I start a review?

File: Packs/Accessdata/Integrations/AccessdataV2/AccessdataV2.py [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+        return _inside
+    return _func
+
+""" register demisto commands """
+
+@wrap_demisto_command("accessdata-api-get-case-by-name")
+def _get_case_by_name(client, name):
+    case = client.cases.first_matching_attribute("name", name)
+    if not case:
+        demisto.error(f"Failed to gather case with name ({name}).")
```
Comment (daryakoval) [inline Packs/Accessdata/Integrations/AccessdataV2/AccessdataV2.py:42]: Change the way you take care of errors. Raise errors in command functions, and catch it at the main function. 
```suggestion
    if not case:
       raise ValueError(f"Failed to gather case with name ({name}).")
```

File: Packs/Accessdata/Integrations/AccessdataV2/AccessdataV2.py [hunk 7]
```diff
@@ -0,0 +1,304 @@
...
+
+@wrap_demisto_command("accessdata-api-create-case")
+def _create_case(client, **kwargs):
+    case = client.cases.create(**kwargs)
+    if not case:
+        demisto.error(f"Failed to create case.")
+        return CommandResults(
+            outputs_prefix="Accessdata.Case",
+            outputs={}
+        )
```
Comment (daryakoval) [inline Packs/Accessdata/Integrations/AccessdataV2/AccessdataV2.py:61]: Change the way you take care of errors. Raise errors in command functions, and catch it at the main function. 
```suggestion
    if not case:
        raise ValueError(f"Failed to create case.")

```

File: Packs/Accessdata/Integrations/AccessdataV2/AccessdataV2.py [hunk 8]
```diff
@@ -0,0 +1,304 @@
...
+    public_certificate = params.get("PUBLIC_CERT", None)
+    client.session.cert = public_certificate
+
+# call function with supplied args
+command = demisto.command()
+func = _run_functions[command]
+args = demisto.args()
+
+# return value from called function
+return_values = func(client, **args)
```
Comment (daryakoval) [inline Packs/Accessdata/Integrations/AccessdataV2/AccessdataV2.py:303]: Surround with try except, and except errors like [here](https://github.com/demisto/content/blob/master/Packs/HelloWorld/Integrations/HelloWorld/HelloWorld.py#L1452).

File: Packs/Accessdata/Integrations/AccessdataV2/AccessdataV2.yml [hunk 9]
```diff
@@ -0,0 +1,242 @@
...
+    outputs:
+    - contextPath: accessdata.case.id
+      description: The identifier of the case.
+      type: integer
+    - contextPath: accessdata.case.name
+      description: The name of the case.
+      type: string
+    - contextPath: accessdata.case.casefolder
+      description: The path to the case data.
+      type: string
```
Comment (daryakoval) [inline Packs/Accessdata/Integrations/AccessdataV2/AccessdataV2.yml:54]: Outputs should be the same as in CommandResults, case sensitive.
```suggestion
    - contextPath: Accessdata.Case.ID
      description: The identifier of the case.
      type: integer
    - contextPath: Accessdata.Case.Name
      description: The name of the case.
      type: string
    - contextPath: Accessdata.Case.CaseFolder
      description: The path to the case data.
      type: string
```

File: Packs/Accessdata/Integrations/AccessdataV2/AccessdataV2.yml [hunk 10]
```diff
@@ -0,0 +1,242 @@
...
+    outputs:
+    - contextPath: accessdata.case.id
+      description: The identifier of the case.
+      type: number
+    - contextPath: accessdata.case.name
+      description: The name of the case.
+      type: string
+    - contextPath: accessdata.case.casefolder
+      description: The path to the case data.
+      type: string
```
Comment (daryakoval) [inline Packs/Accessdata/Integrations/AccessdataV2/AccessdataV2.yml:71]: Please change context to be same as `outputs_prefix` in `CommandResults`: 
```suggestion
    - contextPath: Accessdata.Case.ID
      description: The identifier of the case.
      type: integer
    - contextPath: Accessdata.Case.Name
      description: The name of the case.
      type: string
    - contextPath: Accessdata.Case.CaseFolder
      description: The path to the case data.
      type: string
```

File: Packs/Accessdata/Integrations/AccessdataV2/AccessdataV2.py [hunk 11]
```diff
@@ -0,0 +1,304 @@
...
+    )
+
+_comparator_mapping = {
+    "==": "__eq__",
+    "!=": "__ne__",
+    ">":  "__gt__",
+    "<":  "__lt__",
+    ">=": "__ge__",
+    "<=": "__le__"
+}
```
Comment (daryakoval) [inline Packs/Accessdata/Integrations/AccessdataV2/AccessdataV2.py:208]: Where do you use this?

File: Packs/Accessdata/Integrations/AccessdataV2/AccessdataV2.py (hunk 7 above)
Comment (ThomasVieth) [reply Packs/Accessdata/Integrations/AccessdataV2/AccessdataV2.py]: Code has been updated to make use of Exceptions throughout with try except in the main function body.

File: Packs/Accessdata/Playbooks/AccessDataProcessEvidenceAndExport.yml [hunk 12]
```diff
@@ -0,0 +1,615 @@
+id: AccessData-Process-Evidence
```
Comment (evisochek) [inline Packs/Accessdata/Playbooks/AccessDataProcessEvidenceAndExport.yml:1]: @ThomasVieth  Please run demisto-sdk format on the playbook YML

**Comment by daryakoval** [comment]
@ThomasVieth Have you tested your integration with new docker image? (https://github.com/demisto/content/pull/15060#discussion_r737564211).

In addition please fix validations. (https://github.com/demisto/content/pull/15060#issuecomment-951969121). 

And please change this script https://github.com/demisto/content/pull/15060#discussion_r736570810.

**Comment by daryakoval** [comment]
@ThomasVieth Hi, any updates?

**Comment by daryakoval** [comment]
@ThomasVieth Hi, please see my comments in this pr, update me if you need any help.

**Comment by kgal-pan** [comment]
@daryakoval - @ThomasVieth is out of this project, we're awaiting two new devs to replace him and take over this PR.

**Comment by kgal-pan** [comment]
Closing as Partner has been unresponsive about this.

**Comment by ThomasVieth** [comment]
Hi @kgal-pan @daryakoval Apologies for my absence, have been on site with a customer, however have returned to get this integration completed. Are we able to re-open the PR?

The forked repository has had updates since the last unit-tests have run. This includes updates to the code for Flake8 fulfilment.

**Comment by kgal-pan** [comment]
@ThomasVieth - We cannot reopen this PR since the branch associated with it has already been deleted because the issue became stale. Feel free to create a new one instead.

**Pull request closed without merging**